### PR TITLE
Enable label padding as object

### PR DESCRIPTION
**Pull request by stockiNail**
This PR enables the feature to set the padding option for labels on the annotations by an object.

See CHARTJS doc: https://www.chartjs.org/docs/latest/general/padding.html

`xPadding` and `yPadding` options have been marked as **deprecated**.

### TODO

- [x] test cases
- [x] types definition
- [x] documentation

### CODE REVIEW

**Comment by kurkle** [comment]
I'm a bit hesitant to start merging changes for  version 2 just yet. But otoh it looks like there are no real outstanding bugs to stop that. I'll try to find the time to properly review all of your great work!

**Comment by stockiNail** [comment]
@kurkle Thank you !!! But take your time. I have others tasks/ideas to propose and implement but I don't want to submit too many PRs to avoid you must review tons of things in few time and continuous rebase ;)

**Comment by stockiNail** [comment]
> I'm a bit hesitant to start merging changes for version 2 just yet.

@kurkle I was thinking about that. Let me propose to maintain `xPadding` and `yPadding` (for line annotation) in order to remove the breaking change.
We could add a TODO in the code to remove them when version 2 will be available.
This is the same approach used for `cornerRadius`.
I know that's not elegant but could be a workaround. In this way there PR can be merged and ready for a version 1.2.0.
Make sense?

**Comment by kurkle** [comment]
That would be better in my opinion. We could also add deprecation warnings to make migration easier for users.

**Comment by stockiNail** [comment]
> That would be better in my opinion.

ok, I'm going to change the code in next hours.

> We could also add deprecation warnings to make migration easier for users.

I don't know exactly how it can be implemented in java script. Log message or deprecation annotation somewhere?

**Comment by stockiNail** [comment]
@kurkle xPadding and yPadding has been re-implemented for line annotation.

File: test/fixtures/label/padding.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            },
+            text3: {
+              type: 'label',
+              xScaleID: 'x',
+              yScaleID: 'y',
+              xValue: 'May',
+              yValue: 15,
+              backgroundColor: 'transparent',
+              borderWidth: 1,
+              content: 'position: {x: end}, padding: {y: 20}',
```
Comment (kurkle) [inline test/fixtures/label/padding.js:56]: content does not match with padding

File: test/fixtures/label/padding.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            },
+            text4: {
+              type: 'label',
+              xScaleID: 'x',
+              yScaleID: 'y',
+              xValue: 'February',
+              yValue: 5,
+              backgroundColor: 'transparent',
+              borderWidth: 1,
+              content: 'position: {x: end}, padding: {y: 20}',
```
Comment (kurkle) [inline test/fixtures/label/padding.js:70]: content does not match with padding

**Review by kurkle — COMMENTED** [review]
couple of comments to tests, looking good!

File: test/fixtures/label/padding.js (hunk 2 above)
Comment (stockiNail) [reply test/fixtures/label/padding.js:70]: you're right...

File: test/fixtures/label/padding.js (hunk 1 above)
Comment (stockiNail) [reply test/fixtures/label/padding.js:56]: you're right...

Comment (stockiNail) [reply test/fixtures/label/padding.js]: done

File: test/fixtures/label/padding.js (hunk 2 above)
Comment (stockiNail) [reply test/fixtures/label/padding.js]: done

**Review by kurkle — APPROVED** [review]

**Comment by stockiNail** [comment]
@kurkle THANK YOUUUU!!! I really appreciated!! Tomorrow I'm going to rebase the other PR and fixes my issues.

**Comment by kurkle** [comment]
> @kurkle THANK YOUUUU!!! I really appreciated!! Tomorrow I'm going to rebase the other PR and fixes my issues.

Thank YOU for the hard work!

**Pull request merged**